### PR TITLE
Add deprecation warning for multiple successive semicolons

### PR DESCRIPTION
**Pull request by BioTurboNick**
In preparation for future use of multiple semicolons in an array expression to indicate multi-dimensional concatenation (e.g. #33697), this PR implements a deprecation warning. This syntax is currently allowed but has no effect. Trailing semicolons are still ignored.

```
[1 ; 2] # allowed
[1 ;; 2] == [1 ; 2] # deprecated
```

But, no one would currently see it due to #35648, unless my suggestion #37164 is implemented.

### CODE REVIEW

File: src/julia-parser.scm [hunk 1]
```diff
@@ -1831,7 +1831,15 @@
                          (fix 'vect vec)     ; [x]   => (vect x)
                          (fix 'hcat vec))))  ; [x y] => (hcat x y)
           (case t
-            ((#\; #\newline)
+            ((#\;)
+             (take-token s)
+             (if (eqv? (peek-token s) #\;)
+               (parser-depwarn s (string "Multiple semicolons in an array concatenation expression currently have no effect, "
+                                  "but will map to higher-dimensional concatenation in a future version of Julia.")
```
Comment (JeffBezanson) [inline src/julia-parser.scm:1838]:
```suggestion
                                  "but may have a new meaning in a future version of Julia.")
```

**Comment by JeffBezanson** [comment]
Thanks, looks good to me. This should have a NEWS entry.

**Comment by BioTurboNick** [comment]
@JeffBezanson just saw your suggestion and implemented it. Also added a draft NEWS entry.

**Comment by BioTurboNick** [comment]
I rebased onto the more recent master. Was hoping it would get all tests to pass. I don't think the failures are related though?

**Pull request merged**